src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -201,6 +201,8 @@ private IServiceCollection BuildHostingServices()
             // Conjure up a RequestServices
             services.AddTransient<IStartupFilter, AutoRequestServicesStartupFilter>();
 
+            services.AddTransient<IServiceProviderFactory, DefaultServiceProviderFactory>();
```

Comment:
Can't add it as a field, would break the interface 😄 . Should have made the `IWebHostBuilder` an abstract base class.